src/include/stat.h

```diff
@@ -237,6 +237,20 @@ __wt_stats_clear(void *stats_arg, int slot)
             WT_STAT_SET(session, (session)->dhandle->stats, fld, value);          \
     } while (0)
 
+/*
+ * Update connection and data handle statistics if statistics gathering is enabled. Updates both
+ * statistics concurrently and is useful to avoid the duplicated calls that happen in a lot of
+ * places.
+ */
+#define WT_STAT_CONN_DATA_DECRV(session, fld, value) \
+    WT_STAT_CONN_DECRV(session, fld, value);         \
```

Comment:
Hmm, I'm a bit concerned with this now that I think about it.

I think it might need to be wrapped in a `do...while` loop like some of the other macros. The reason why is because if you write something like:

```
if (blah)
    WT_STAT_CONN_DATA_DECRV(session, stat_1, 1);
```

That will expand to this after the pre-processor is done with it (which would be unexpected):

```
if (blah)
    WT_STAT_CONN_DECRV(session, stat_1, 1);
WT_STAT_DATA_DECRV(session, stat_1, 1);
```

Reply:
@agorrod Do you agree that this should be nested in a loop?

Reply:
These absolutely need to be within a do...while(0) loop. Good catch Alex.

Reply:
Ah, yep understood! Thanks Alex and Sue for pointing that out, will get that fixed and keep an eye out for this in the future.